plugins/experimental/txn_box/plugin/CMakeLists.txt

```diff
@@ -0,0 +1,62 @@
+#######################
+#
+#  Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+#  agreements.  See the NOTICE file distributed with this work for additional information regarding
+#  copyright ownership.  The ASF licenses this file to you under the Apache License, Version 2.0
+#  (the "License"); you may not use this file except in compliance with the License.  You may obtain
+#  a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software distributed under the License
+#  is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing permissions and limitations under
+#  the License.
+#
+#######################
+cmake_minimum_required(VERSION 3.11)
+
+project(txn_box)
+set(CMAKE_CXX_STANDARD 17)
+
+pkg_check_modules(PCRE2 REQUIRED IMPORTED_TARGET libpcre2-8)
```

Comment:
This is checked at the top level now

Reply:
if I don't use this, then cmake is not able to find pcre:
```
CMake Error at cmake/add_atsplugin.cmake:21 (add_library):
  Target "txn_box" links to target "PkgConfig::PCRE2" but the target was not
  found.  Perhaps a find_package() call is missing for an IMPORTED target, or
  an ALIAS target is missing?
Call Stack (most recent call first):
  plugins/experimental/txn_box/plugin/CMakeLists.txt:20 (add_atsplugin)
```
Earlier I've got this:
```
CMake Warning at CMakeLists.txt:261 (find_package):
  By not providing "FindPCRE2.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "PCRE2", but
  CMake did not find one.

  Could not find a package configuration file provided by "PCRE2" with any of
  the following names:

    PCRE2Config.cmake
    pcre2-config.cmake

  Add the installation prefix of "PCRE2" to CMAKE_PREFIX_PATH or set
  "PCRE2_DIR" to a directory containing one of the above files.  If "PCRE2"
  provides a separate development package or SDK, be sure it has been
  installed.
```
wondering if I am missing some pcre2 package.

Reply:
oh yeah, the check at the top level is not correct.  We can clean this up later.